bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/graphics/ImageLoader.java

```diff
@@ -155,14 +155,19 @@ public ImageData[] load(InputStream stream) {
 	return data;
 }
 
-List<ElementAtZoom<ImageData>> load(InputStream stream, int fileZoom, int targetZoom) {
+List<ImageData> load(InputStream stream, int fileZoom, int targetZoom) {
 	if (stream == null) SWT.error(SWT.ERROR_NULL_ARGUMENT);
 	reset();
-	List<ElementAtZoom<ImageData>> images = NativeImageLoader.load(new ElementAtZoom<>(stream, fileZoom), this, targetZoom);
-	data = images.stream().map(ElementAtZoom::element).toArray(ImageData[]::new);
+	List<ImageData> images = NativeImageLoader.load(new ElementAtZoom<>(stream, fileZoom), this, targetZoom);
+	data = images.stream().toArray(ImageData[]::new);
 	return images;
 }
 
+boolean canLoadAtZoom(InputStream stream, int fileZoom, int targetZoom) {
```

Comment:
This method could be static, which would avoid the creation of otherwise unused `ImageLoader` instances.
```suggestion
static boolean canLoadAtZoom(InputStream stream, int fileZoom, int targetZoom) {
```

Reply:
Makes sense, adapted accordingly.